cli/lib/nf3.mjs

```diff
@@ -292,10 +244,12 @@ class Nf3 {
         to: contractAddress,
         data: unsignedTransaction,
         value: fee,
-        gas,
-        gasPrice,
         nonce: this.nonce,
       };
+      // Estimate the gasLimit
+      const gas = await estimateGas(tx, this.web3);
```

Comment:
Don’t we need gasPrice as well?

Reply:
@druiz0992 

A quick search revealed that gas calculations were outdated due to London Upgrade of August 2021. Quick summary [1]:

**Prior to the London upgrade**

```
Total fee = Gas units (limit) * Gas price per unit
```

**After the London upgrade**

```
Total fee = Gas units * (base fee + priority fee)
```

_Where the base fee is a value set by the protocol and the priority fee is a value set by the user as a tip to the validator._

**What this means re transaction object**

Before we had `gas` and `gasPrice`. Now you can set [2]:

- `maxFeePerGas`: (optional, defaulted to (`2 * block.baseFeePerGas) + maxPriorityFeePerGas`) The maximum fee per gas that the transaction is willing to pay in total

- `maxPriorityFeePerGas`: (optional, defaulted to 2.5 Gwei) The maximum fee per gas to give miners to incentivise them to include the transaction (Priority fee)

**About this PR**

Limits gas calculations to `estimateGas` function via web3 - this is for calculating "gas units", a.k.a. `gas`.
In the future, we can consider using dynamic values for `maxPriorityFeePerGas` and `maxFeePerGas`.

Reply:
**References**

[1] https://ethereum.org/en/developers/docs/gas
[2] https://web3js.readthedocs.io/en/v1.7.3/web3-eth.html#id88

Reply:
i still seem to remember that when we deployed last time to testnet (beginning of november), we had to update gasPrice because otherwise it wouldn't work.
```
WARN [10-26|14:36:39.709] Served eth_call                          reqid=22578            duration="199.712µs"  err="err: max fee per gas less than block base fee: address 0xe327221Fc90f4D938E23A03C58a4e93A66AAEC20, maxFeePerGas: 20000000000 baseFee: 186671733869 (supplied gas 8000000)"
```